AndroidPubSub/README.md

```diff
@@ -149,3 +149,71 @@ The keytool command does not allow importing an existing private key into a keys
         adb push <keystore name>.bks /data/user/0/your_app_dir_goes_here/files/<keystore name>
 
     The directory and filename used will depend on your use case.  Typically the application's files directory is in /data/user/0/<app namespace>/files/.  You may however choose to locate your keystore on removable media or another space on the filesystem.  The SDK allows for specifying the file path and name of the keystore so the choice is up to you.
+	
+## Using AWS IoT with authenticated users
+
+This sample demonstrates enabling unauthenticated users to use the AWS IoT APIs to securely publish-to and subscribe-from MQTT topics. Following steps demonstrates usage of AWS IoT with authenticated users. 
+
+1. Follow the steps [here](https://docs.aws.amazon.com/aws-mobile/latest/developerguide/how-to-user-sign-in-sign-out.html) to add authentication UI to your app.
+
+2. Get the credentials provider
+
+```
+credentialsProvider =
+               IdentityManager.getDefaultIdentityManager().getCredentialsProvider();
+```
+
+3. Get CognitoIdentity from credentialsProvider and use it to attach pricipal policy to the IotAndroidClient. Create IoT policy as specified in step 4 of [Using the sample](https://github.com/awslabs/aws-sdk-android-samples/tree/desokroshan-patch-2/AndroidPubSub#using-the-sample) section above. 
+
+```
+// Setup the logins map
+String token = IdentityManager.getDefaultIdentityManager().getCurrentIdentityProvider().getToken();
```

Comment:
IdentityProvider currentIdentityProvider = identityManager.getCurrentIdentityProvider();
String token = currentIdentityProvider.getToken();
String providerKey = currentIdentityProvider.getCognitoLoginKey();
logins.put(providerKey, token);

Reply:
This will reduce the need for the customer to have to search or construct objects.